pyro/nn/module.py

```diff
@@ -136,24 +169,30 @@ def forward(self):
         object.
     """
 
-    def __init__(self, prior):
+    prior: Union[
+        "TorchDistributionMixin", Callable[["PyroModule"], "TorchDistributionMixin"]
+    ]
+
+    def __post_init__(self) -> None:
         super().__init__()
```

Comment:
I believe `super().__init__()` should be omitted because it should be called in the autogenerated `__init__` method.

Reply:
https://docs.python.org/3/library/dataclasses.html#post-init-processing says that `super().__init__()` should be called by the `__post_init__`. But since `PyroSample` isn't subclassing another class there is no need to call `super().__init__()`, right? So I removed it anyways.